app/Support/SearchIndexBuilder.php

```
@@ -0,0 +1,408 @@
+        }
+        return array_values($ids);
+    }
+
+    /**
+     * Rebuild a batch of books. Accepts a pre-captured id list (see
+     * bookIdsForAuthor/bookIdsForPublisher) so merge/delete callers can snapshot
+     * the affected set before the linking rows disappear.
+     *
+     * @param int[] $bookIds
+     */
+    public static function rebuildMany(\mysqli $db, array $bookIds): void
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $db. Configured minimum length is 3. (undefined)

(ShortVariable)